classes/04class/exercises/c04-iac02/ramiray/vpc.tf

```diff
@@ -0,0 +1,139 @@
+#
+# VPC resources
+#
+resource "aws_vpc" "da_iac02_vpc" {
+  cidr_block           = var.cidr_block
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+
+  tags = merge(
+    {
+      Name        = var.name
+        },
+    var.tags
+  )
+}
+
+resource "aws_internet_gateway" "da_iac02_igw" {
+  vpc_id = aws_vpc.da_iac02_vpc.id
+
+  tags = merge(
+    {
+      Name        = "gwInternet"
+        },
+    var.tags
+  )
+}
+
+resource "aws_route_table" "private" {
+  count = length(var.private_subnet_cidr_blocks)
+
+  vpc_id = aws_vpc.da_iac02_vpc.id
+
+  tags = merge(
+    {
+      Name        = "PrivateRouteTable"
+                 },
+    var.tags
+  )
+}
+
+resource "aws_route" "private" {
+  count = length(var.private_subnet_cidr_blocks)
+
+  destination_cidr_block  = var.destination_cidr_block 
+  route_table_id         = aws_route_table.private[count.index].id
+   nat_gateway_id         = aws_nat_gateway.iac02nat[count.index].id
+ }
+
+resource "aws_route_table" "public" {
+  vpc_id = aws_vpc.da_iac02_vpc.id
+
+  tags = merge(
+    {
+      Name        = "PublicRouteTable"
+     },
+    var.tags
+  )
+}
+
+resource "aws_route" "public" {
+  route_table_id         = aws_route_table.public.id
+ #count = length(var.public_subnet_cidr_blocks)
+  gateway_id             = aws_internet_gateway.da_iac02_igw.id
+  #nat_gateway_id         = aws_nat_gateway.iac02nat.id
+  destination_cidr_block  = var.destination_cidr_block 
+}
+
+resource "aws_subnet" "private" {
+  count = length(var.private_subnet_cidr_blocks)
+
+  vpc_id            = aws_vpc.da_iac02_vpc.id
+  cidr_block        = var.private_subnet_cidr_blocks[count.index]
+  availability_zone = var.availability_zones[count.index]
```

Comment:
Just be careful with that, it might break your code if the length between `availability_zones` and `private_subnet_cidr_blocks` is different.
A good way to avoid that it to use `availability_zones` as your length size and use [`cidrsubnet`](https://www.terraform.io/docs/configuration/functions/cidrsubnet.html) to calculate your subnet cidr_block based on the vpc_cidr_block.

Reply:
Hi Alan, 

Thanks for your tips and you are doing great in reviewing our exercises.

Reply:
Thanks mate,
Happy to help 😁😁
I am learning a lot from you guys as well

Let me know if you need something